chrome/content/zotero/feedSettings.js

```diff
@@ -0,0 +1,172 @@
+/*
+    ***** BEGIN LICENSE BLOCK *****
+    
+    Copyright © 2015 Center for History and New Media
+                     George Mason University, Fairfax, Virginia, USA
+                     http://zotero.org
+    
+    This file is part of Zotero.
+    
+    Zotero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    
+    Zotero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+    
+    You should have received a copy of the GNU Affero General Public License
+    along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+    
+    ***** END LICENSE BLOCK *****
+*/
+
+//////////////////////////////////////////////////////////////////////////////
+//
+// Zotero_Feed_Settings
+//
+//////////////////////////////////////////////////////////////////////////////
+
+var Zotero_Feed_Settings = new function() {
+	let urlIsValid = true,
+		data = null,
+		feedReader = null,
+		urlTainted = false;
+	
+	let cleanURL = function(url) {
+		let cleanUrl = Zotero.Utilities.cleanURL(url, true);
+
+		if (cleanUrl) {
+			if (/^https?:\/\/[^\/\s]+\/\S/.test(cleanUrl)) {
+				return cleanUrl;
+			} else {
+				Zotero.debug(uri.scheme + " is not a supported protocol for feeds.");
+			}
+		}
+	};
+	
+	this.init = function() {
+		this.toggleAdvancedOptions(false);
+		
+		data = window.arguments[0];
+		
+		if (data.url) {
+			document.getElementById('feed-url').value = data.url;
+		}
+		
+		if (!data.url) {
+			this.invalidateUrl();
+		} else {
+			// Do not allow to change URL for existing feed
+			document.getElementById('feed-url').readOnly = true;
+		}
+		
+		if (data.title) {
+			document.getElementById('feed-title').value = data.title;
+		}
+		
+		let ttl;
+		if (data.ttl !== undefined) {
+			ttl = Math.floor(data.ttl / 60);
+		} else {
+			ttl = 1;
+		}
+		document.getElementById('feed-ttl').value = ttl;
+		
+		let cleanupAfter = data.cleanupAfter;
+		if (cleanupAfter === undefined) cleanupAfter = 2;
+		document.getElementById('feed-cleanupAfter').value = cleanupAfter;
+		
+		if (data.url && !data.urlIsValid) {
+			this.validateUrl();
```

Comment:
Usually better not to leave async functions non`yield`ing — something could be added later below this call that expects the URL to be validated already. Better to make use `coroutine` on `init` and add a `yield` here.